tests/tests_pytorch/callbacks/test_deepspeed_model_summary.py

```diff
@@ -0,0 +1,71 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytorch_lightning as pl
+from pytorch_lightning import Callback, Trainer
+from pytorch_lightning.callbacks import DeepSpeedModelSummary
+from pytorch_lightning.callbacks.deepspeed_model_summary import DeepSpeedSummary
+from pytorch_lightning.demos.boring_classes import BoringModel
+from pytorch_lightning.strategies import DeepSpeedStrategy
+from tests_pytorch.helpers.runif import RunIf
+
+
+@RunIf(min_cuda_gpus=1, deepspeed=True)
+def test_deepspeed_default_summary(tmpdir):
+    """Test to ensure that the trainer enables the DeepSpeedSummary when DeepSpeed is used."""
+
+    trainer = Trainer(
+        strategy=DeepSpeedStrategy(),
+        default_root_dir=tmpdir,
+        accelerator="gpu",
+        fast_dev_run=True,
+        devices=1,
+        max_epochs=2,
+        precision=16,
+        enable_model_summary=True,
+    )
+
+    assert DeepSpeedModelSummary in [type(cb) for cb in trainer.callbacks]
+
+
+@RunIf(min_cuda_gpus=2, deepspeed=True, standalone=True)
+def test_deepspeed_summary_values(tmpdir):
+    """Test to ensure that the summary contains the correct values when stage 3 is enabled."""
+
+    model = BoringModel()
+    total_parameters = sum(x.numel() for x in model.parameters())
+
+    class TestCallback(Callback):
+        def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+            model_summary = DeepSpeedSummary(pl_module, max_depth=1)
```

Comment:
Why not get the existing modelsummary from the Trainer callbacks?

Reply:
What do you mean? This is a bit hacky but I just want to ensure that the overridden ModelSummary object works as intended. 

There is a bit of naming confusion here (ModelSummary is both a callback, and a class defined in utilities).